core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -735,7 +749,12 @@ object SparkSubmit extends CommandLineUtils {
     }
 
     try {
-      mainMethod.invoke(null, childArgs.toArray)
+      if (isSparkApp) {
+        val envvars = Map[String, String]() ++ sys.env
+        mainMethod.invoke(null, childArgs.toArray, childSparkConf, envvars.toMap)
```

Comment:
ah, thanks for catching this. everything should be passed in the confs now and we've deprecated all the env configs